captum/attr/_core/deep_lift.py

```diff
@@ -174,6 +176,20 @@ def attribute(
                         is set to True convergence delta will be returned in
                         a tuple following attributions.
                         Default: False
+            custom_attribution_func (callable, optional): A custom function for
+                        computing final attribution scores. This function can take
+                        at least one and at most three arguments with the
+                        following signature:
+                            - custom_attribution_func(multipliers)
+                            - custom_attribution_func(multipliers, inputs)
+                            - custom_attribution_func(multipliers, inputs, baselines)
+                        In case this function is not provided we use the default
+                        logic defined as: multipliers * (inputs - baselines)
+                        It is assumed that `custom_attribution_func` returns a
+                        tuple of attribution tensors that have the same shape and
+                        dimensionality as the `inputs`.
+
```

Comment:
It might also be worth documenting that the deltas may not be meaningful depending on this function, e.g. with a hypothetical input, since the delta should then be based on the hypothetical input evaluation (and even then, the completeness property would also not necessarily hold)? We could potentially allow this function to return overriden inputs and baselines to compute a delta based on that, but that may be unnecessarily complicated, and a user could do it themselves later if they really want it with compute_convergence_delta. Alternatively, we could just disallow delta computation in conjunction with custom_attribution_func and just return the output of the function, which lets the user do whatever they want with the multiplier, inputs, and baselines. What do you think?

Reply:
Good point! I think we can leave the logic of computing deltas and add documentation about convergence deltas. 
Too many checks will make the code harder to maintain.